File: SDK/DotNet/Uloq.SDK/Uloq.SDK/QR/QRGenerator.cs
using System.Threading.Tasks;
using Uloq.SDK.Connectors;
using Uloq.SDK.Eccenscia.Services.Models.UloqRequestor;
using Uloq.SDK.Models;

namespace Uloq.SDK.QR
{
    public class QRGenerator
    {
        private readonly ConnectionModel _connectionModel;

        public QRGenerator(ConnectionModel connectionModel)
        {
            _connectionModel = connectionModel;
        }

        /// <summary>
        /// Generate the QR Code from the Uloq services
        /// </summary>
        /// <param name="jsonPayload"></param>
        /// <returns>QRcodeResponse</returns>
        public async Task<QRCodeResponse> GenerateQRCode(QRCodeRequest request)
        {
            HttpConnector httpConnector = new HttpConnector(_connectionModel);
            OutcomeObject<QRCodeResponse> result = await httpConnector.PostAsync<QRCodeResponse>("createqrcode", request.ToJson());

            if (result.Success)            
                return result.Result;            
            
            return null;
        }
    }
}

File: SDK/Python/uloq_sdk/my_module/QRGenerator.py

import asyncio
from typing import Optional
from .connectors import HttpConnector
from .models import QRCodeRequest, QRCodeResponse, ConnectionModel, OutcomeObject

class QRGenerator:
    def __init__(self, connection_model: ConnectionModel):
        self._connection_model = connection_model

    async def generate_qr_code(self, request: QRCodeRequest) -> Optional[QRCodeResponse]:
        http_connector = HttpConnector(self._connection_model)
        result = await http_connector.post_async("createqrcode", request.to_json())
        
        if result.success:
            return result.result
        return None
